pydra/engine/core.py

```diff
@@ -973,15 +988,21 @@ def set_output(self, connections):
         logger.info("Added %s to %s", self.output_spec, self)
 
     def _collect_outputs(self):
+        error = False
         output_klass = make_klass(self.output_spec)
         output = output_klass(**{f.name: None for f in attr.fields(output_klass)})
         # collecting outputs from tasks
         output_wf = {}
         for name, val in self._connections:
             if not isinstance(val, LazyField):
                 raise ValueError("all connections must be lazy")
-            output_wf[name] = val.get_value(self)
-        return attr.evolve(output, **output_wf)
+            val_out = val.get_value(self)
+            if val_out == "error":
+                error = True
```

Comment:
why not just raise a `ValueError` here instead of using an `error` variable?

Reply:
yes, I should change